### PR TITLE
Feat/plugin abstraction

### PR DESCRIPTION
**Pull request by jaredcbaum**

### CODE REVIEW

File: src/theme/DocItem/index.tsx [hunk 1]
```diff
@@ -5,28 +5,28 @@
...
+ import DocVersionBadge from '@theme/DocVersionBadge';
+ import Seo from '@theme/Seo';
+ import type {Props} from '@theme/DocItem';
+ import DocItemFooter from '@theme/DocItemFooter';
+ import TOC from '@theme/TOC';
+ import TOCCollapsible from '@theme/TOCCollapsible';
+ import Heading from '@theme/Heading';
+ import styles from './styles.module.css';
+ import {ThemeClassNames, useWindowSize} from '@docusaurus/theme-common';
+ import DocBreadcrumbs from '@theme/DocBreadcrumbs';
```
Comment (ltm) [inline src/theme/DocItem/index.tsx:21]: There's a bunch of leading spaces here:
```suggestion
import React from 'react';
import clsx from 'clsx';
import DocPaginator from '@theme/DocPaginator';
import DocVersionBanner from '@theme/DocVersionBanner';
import DocVersionBadge from '@theme/DocVersionBadge';
import Seo from '@theme/Seo';
import type {Props} from '@theme/DocItem';
import DocItemFooter from '@theme/DocItemFooter';
import TOC from '@theme/TOC';
import TOCCollapsible from '@theme/TOCCollapsible';
import Heading from '@theme/Heading';
import styles from './styles.module.css';
import {ThemeClassNames, useWindowSize} from '@docusaurus/theme-common';
import DocBreadcrumbs from '@theme/DocBreadcrumbs';
```

**Comment by ltm** [comment]
The sidebar is offset incorrectly:
Screen Shot 2022-03-02 at 4 03 02 PM

**Comment by jaredcbaum** [comment]
> The sidebar is offset incorrectly:
> Screen Shot 2022-03-02 at 4 03 02 PM

Yeah some styles in the sidebar broke on the update to the latest version. It's on my radar 👍

File: src/theme/DocItem/index.tsx (hunk 1 above)
Comment (jaredcbaum) [reply src/theme/DocItem/index.tsx]: Good catch! fixed

**Pull request closed without merging**